### PR TITLE
Fenicsx v0.5.0

### PR DESCRIPTION
**Pull request by ma595**
Update fenicsx packages to latest tagged releases (v0.5.0).

### CODE REVIEW

**Review by nate-sime — DISMISSED** [review]

File: var/spack/repos/builtin/packages/fenics-basix/package.py [hunk 1]
```diff
@@ -15,11 +15,21 @@ class FenicsBasix(CMakePackage):
...
+        "%gcc@:9.10", when="@0.5.0:", msg="fenics-basix requires GCC-10 or newer for C++20 support"
+    )
+    conflicts(
+        "%clang@:9.10",
+        when="@0.5.0:",
+        msg="fenics-basix requires Clang-10 or newer for C++20 support",
+    )
+
     depends_on("cmake@3.18:", type="build")
     depends_on("xtl@0.7.2:")
```
Comment (garth-wells) [inline var/spack/repos/builtin/packages/fenics-basix/package.py]: Basix v0.5 no longer depends on `xtl` or `xtensor`.

File: var/spack/repos/builtin/packages/py-fenics-basix/package.py [hunk 2]
```diff
@@ -16,12 +16,14 @@ class PyFenicsBasix(PythonPackage):
...
 
     version("main", branch="main")
+    version("0.5.0", sha256="7cd5a8b053219941e84068731b88338002f1f09dfc73dcd41f9cd584588dc65a")
     version("0.4.2", sha256="a54f5e442b7cbf3dbb6319c682f9161272557bd7f42e2b8b8ccef88bc1b7a22f")
     version("0.3.0", sha256="9b148fd2a5485c94011fc6ca977ebdef0e51782a62b3654fc044f35b60e2bd07")
     version("0.2.0", sha256="e1ec537737adb283717060221635092474e3f2b5b5ba79dfac74aa496bec2fcb")
     version("0.1.0", sha256="2ab41fe6ad4f6c42f01b17a6e7c39debb4e0ae61c334d1caebee78b741bca4e7")
 
     depends_on("fenics-basix@main", type=("build", "run"), when="@main")
+    depends_on("fenics-basix@0.5.0", type=("build", "run"), when="@0.5.0")
```
Comment (garth-wells) [inline var/spack/repos/builtin/packages/py-fenics-basix/package.py]: v0.5 no longer depends on `xtl` or `xtensor`.

**Review by garth-wells — CHANGES_REQUESTED** [review]
Looks good, I've left some inline comments.

`(py)-fenics-dolfinx@main` no longer depends on `xtl` and `xtensor`.

Has this PR been tested using the CI at https://github.com/FEniCS/dolfinx/actions/workflows/spack.yml?

**Review by garth-wells — APPROVED** [review]

**Review by chrisrichardson — APPROVED** [review]

**Review by nate-sime — APPROVED** [review]

**Comment by garth-wells** [comment]
@tldahlgren I think this is ready to merge now.

**Comment by jhale** [comment]
I have a slightly refined version of this patch that better reflects some upcoming changes on `main` and marks some of the older versions as deprecated which will allow us to reduce complexity of these files in the future. I will push it to a new PR shortly.

**Review by tldahlgren — APPROVED** [review]
Confirmed all new sha256.

**Pull request merged**